src/loader/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.4)
```

Comment:
I don't think you need this in a sub-cmakelist file? If it is required, we should use a variable for the version number to ensure versions requirements are in-sync across `CMakeLists.txt` files.